projects/admin/src/app/menu/service/library-switch.service.ts

```diff
@@ -70,6 +73,9 @@ export class LibrarySwitchService {
     }
     // Update current library on user
     user.setCurrentLibrary(libraryPid);
+    this._libraryApiService.getClosedDates(libraryPid).subscribe(
```

Comment:
Is it really the right place? It would be better to observe a switch of library and the date component can do the job, unless it is used at several places.

Reply:
This is the place where the library is changed. When the library change, I call the API and emit closedDate when API call is success.